components/MobileNavigation.js

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from 'react'
+import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup'
+
+import LevitatingLink from './LevitatingLink'
+import Hamburger from '../resources/hamburger.svg'
+
+export default class MobileNavigation extends Component {
+  state = {
+    showMobileNav: false,
+    expandMobileNav: false
+  }
+
+  componentDidMount () {
+    window.addEventListener('scroll', this.handleScroll)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('scroll', this.handleScroll)
+  }
+
+  handleScroll = (e) => {
+    window.scrollY > 700 ? this.showMobileNav() : this.hideMobileNav()
+  }
+
+  showMobileNav = () => {
+    this.setState({
+      showMobileNav: true
+    })
+  }
+  hideMobileNav = () => {
+    this.setState({
+      showMobileNav: false,
+      expandMobileNav: false
+    })
+  }
+
+  handleToggleMobileNav = (value) => {
+    this.setState({
+      expandMobileNav: value || !this.state.expandMobileNav
+    })
+  }
+
+  handleClickNav = (e) => {
+    this.handleToggleMobileNav(false)
+  }
+
+  render () {
+    const { showMobileNav, expandMobileNav } = this.state
+
+    return (
+      <CSSTransitionGroup
+        transitionName='fade'
+        transitionEnterTimeout={500}
+        transitionLeaveTimeout={500}>
+        {
+          showMobileNav &&
+            <div id='mobile-nav'>
+              <button type='button' className='btn-nav-toggle' onClick={() => this.handleToggleMobileNav()}>
+                <Hamburger />
+              </button>
+              <CSSTransitionGroup
+                transitionName='fade'
+                transitionEnterTimeout={500}
+                transitionLeaveTimeout={300}>
+                {
+                  expandMobileNav &&
+                    <nav onClick={this.handleClickNav}>
+                      <div className='nav-header'>React Bangkok</div>
+                      <NavigationLink href='#about'>About</NavigationLink>
```

Comment:
These lines are duplicated form MainNav. Can we somehow de-duplicate it? So that when more sections get added, we can change the code in one place, eliminating the possibility of both menus going inconsistent.